src/segmented_buffer.rs

```diff
@@ -263,24 +379,47 @@ where
     }
 }
 
-impl<F> SegmentedPoolBuf<F, BytesMut, AllocBytesMutFn> {
-    pub fn into_bytes_stream(self) -> SegmentedBufBytes {
+impl<F> SegmentedPoolBuf<F, Buffer, AllocBufferFn> {
+    pub fn into_bytes_stream(mut self) -> SegmentedBufBytes {
```

Comment:
This is not used within the client or the agent.

Reply:
And it's gone, which let me simplify the Buffer further